tests/dogtag/pytest-ansible/pytest/performance_test/test_cert_enrollment.py

```diff
@@ -71,32 +77,51 @@ def __init__(self, sn_uid):
 
         enrollment_results = cert_client.enroll_cert('caUserCert', inputs)
 
+        if enrollment_results.__len__() == 0:
+            raise Exception("Enrollment results is empty")
+
         for enrollment_result in enrollment_results:
             request_data = enrollment_result.request
             cert_data = enrollment_result.cert
-            print('Request ID: ' + request_data.request_id)
+            if request_data.request_status != 'complete':
+                raise Exception("Cert enrollment failed : {}".format(cert_data))
+            else:
+                self.cert_serial_number=cert_data.serial_number
+
 
 def run_test(sn_uid, number_of_tests_per_thread):
     # execute the specified number of tests sequentially
     for i in range(number_of_tests_per_thread):
-        cert_enroll(sn_uid)
+        serial_num=cert_enroll(sn_uid)
+        cert_list.append(serial_num.cert_serial_number)
     # call cert_enroll(sn_uid)
 
 
 if __name__ == "__main__":
     # get test parameters from CLI parameters
-    number_of_threads = args.number_of_threads
-    number_of_tests_per_thread = args.number_of_tests_per_thread
+    number_of_threads = args.number_of_clients
+    number_of_tests_per_thread = args.number_of_tests_per_client
 
     # create the specified number of threads
     threads = []
-
+    cert_list=[]
     for t in range(number_of_threads):
         t1 = threading.Thread(target=run_test, args=("testuser{}".format(t), number_of_tests_per_thread))
         t1.start()
         threads.append(t1)
 
     # wait for all threads to complete
-    # (not sure if this is needed since Python probably won’t exit until all threads are complete)
     for t in threads:
         t.join()
+    end = timer()
+
+    cf=open("cert_file.txt", "w+")
+    pickle.dump(cert_list, cf)
```

Comment:
@edewata Are we fine with pickle here? Or would json suffice?

Reply:
The file at least initially will just be shared among the performance test scripts, so any format should be fine as long as it's consistent. If JSON provides some other benefits we can certainly use that too.

Reply:
Update with JSON instead of pickle